restli-int-test/src/test/java/com/linkedin/restli/examples/TestFillInDefaultValueFooDefault.java

```diff
@@ -0,0 +1,128 @@
+package com.linkedin.restli.examples;
+
+import com.linkedin.data.DataList;
+import com.linkedin.r2.RemoteInvocationException;
+import com.linkedin.restli.client.FindRequest;
+import com.linkedin.restli.client.GetAllRequest;
+import com.linkedin.restli.common.CollectionMetadata;
+import com.linkedin.restli.common.CollectionResponse;
+import com.linkedin.restli.common.Link;
+import com.linkedin.restli.common.LinkArray;
+import com.linkedin.restli.common.RestConstants;
+import com.linkedin.restli.examples.defaults.api.FooDefault;
+import com.linkedin.restli.examples.defaults.api.FooDefaultFillingRequestBuilders;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+
+public class TestFillInDefaultValueFooDefault extends RestLiIntegrationTest
+{
+  @BeforeClass
+  public void initClass() throws Exception
+  {
+    super.init(Collections.emptyList());
+  }
+
+  @AfterClass
+  public void shutDown() throws Exception
+  {
+    super.shutdown();
+  }
+
+  @DataProvider(name = "testFinderData")
+  private Object[][] testFinderData() throws CloneNotSupportedException
+  {
+    List<FooDefault> expectedElements = new ArrayList<>();
+    int count = 2;
+    for (int i = 0; i < count; i++)
+    {
+      FooDefault obj = new FooDefault().setId(i).setFooDefaultId(0);
+      expectedElements.add(obj);
+    }
+
+    CollectionMetadata expectedCollectionMetadata = new CollectionMetadata();
+    expectedCollectionMetadata.setCount(count);
+    expectedCollectionMetadata.setStart(0);
+    Link expectedLink = new Link();
+    expectedLink.setRel("next");
+    expectedLink.setType("application/json");
+    expectedLink.setHref("/fooDefaultFilling?$fillInDefaults=true&q=findFooCollection&start=2&count=2");
+    LinkArray expectedLinkArray = new LinkArray();
+    expectedLinkArray.add(expectedLink);
+    expectedCollectionMetadata.setLinks(expectedLinkArray);
+    expectedCollectionMetadata.setTotal(0);
+
+    FooDefault expectedCustomMetadata = new FooDefault();
+    expectedCustomMetadata.setFooDefaultId(0);
+    expectedCustomMetadata.setId(count);
+    return new Object[][] {{2, expectedElements, expectedCollectionMetadata, expectedCustomMetadata}};
+  }
+
+  @Test(dataProvider = "testFinderData")
+  public void testFinder(int param,
+      List<FooDefault> expectedElements,
+      CollectionMetadata expectedCollectionMetadata,
+      FooDefault expectedCustomMetadata) throws RemoteInvocationException
+  {
+    FooDefaultFillingRequestBuilders requestBuilders = new FooDefaultFillingRequestBuilders();
+    FindRequest<FooDefault> request = requestBuilders.findByFindFooCollection()
+        .setParam("count", param)
+        .setParam(RestConstants.FILL_IN_DEFAULTS_PARAM, true)
```

Comment:
ideally we should test both with true and false and verify the change in behavior. The current tests will not catch the case when server code is directly setting the field and the "fillInDefault" behavior is broken.

Reply:
Thanks, I added a test for not having the param that set to fill in default values in the test of get_all